cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,13 +186,28 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      final List<String> additionalClassPaths = new ArrayList<>();
+      List<String> newExtraJars = new ArrayList<>();
+      // Add extra jars set in cConf to additionalClassPaths and localizeResources
+      for (URI jarURI : CConfigurationUtil.getExtraJars(cConf)) {
+        LocalizeResource localizeResource;
+        String protocol = jarURI.getScheme();
+        localizeResource = new LocalizeResource(jarURI, false);
+        String jarName = Networks.getURIFilename(jarURI);
+        localizeResources.put(jarName, localizeResource);
+        additionalClassPaths.add(jarName);
+        String jarPath = (protocol == null || "file".equals(protocol)) ? jarName : jarURI.toString();
```

Comment:
The URIs returned by the `getExtraJars` call already guaranteed that protocol won't be `null`, isn't it?